### PR TITLE
SQLite-backed DOs: check actor brokenness before starting implicit transactions

### PR DESCRIPTION
**Pull request by jclee**

### CODE REVIEW

File: src/workerd/io/actor-sqlite.c++ [hunk 1]
```diff
@@ -198,6 +198,7 @@ void ActorSqlite::ExplicitTxn::rollbackImpl() noexcept(false) {
 
 void ActorSqlite::onWrite() {
   if (currentTxn.is<NoTxn>()) {
+    requireNotBroken();
```
Comment (kentonv) [inline src/workerd/io/actor-sqlite.c++:201]: Let's move this above the `if()`, so that all writes will fail when broken.

**Comment by jclee** [comment]
Tried using the `fixup.sh` script, but it pushed a rebase I didn't expect...  At any rate, here were the changes in its output:
* 87bd15f2
* 64ff1c5b

File: src/workerd/io/actor-sqlite.c++ (hunk 1 above)
Comment (jclee) [reply src/workerd/io/actor-sqlite.c++]: I moved the line, and also added a rudimentary test.

**Review by kentonv — APPROVED** [review]

**Comment by jclee** [comment]
Also adding an extra check for brokenness to the `getSqliteDatabase()` function, to reduce the possibility of read-only queries succeeding against a broken actor.

**Comment by jclee** [comment]
Also fixing lint.

**Comment by jclee** [comment]
Dropped the check for brokenness in the getSqliteDatabase() function... it causes an internal test to hang in a way I don't fully understand yet.  I probably just need to update the test to match, but want to get this in first.

**Pull request merged**